MVP of issue #3630:  Search: Ability to pass a custom SearchCategory renderer to the Search component

alexasselin008 opened the pull request.

PrincessMadMath committed:
[WIP] First implementation of custom SearchCategoryRenderer

alexasselin008 committed:
[WIP] Added Typescript file and props validation

alexasselin008 committed:
[wip] updating documentation

Commit:
fix(Dropdown): search does not preserve value (#3584)

* Fix.

* add UT to cover case

layershifter committed:
docs(Examples): reduce number of items in examples, fix HMR issues (#3651)

Commit:
feat(Dropdown): add 'closeOnEscape' prop (#3632)

* feat(Dropdown): add 'closeOnEscape' prop (#3622)

Signed-off-by: mhatvan <markus_hatvan@aon.at>

* Update Dropdown-test.js

alexasselin008 committed:
WIP  remove unecessary change in PR

patricklafrance commented on docs/src/examples/modules/Search/Types/SearchExampleCategoryCustom.js:
```diff
@@ -4,6 +4,20 @@ import faker from 'faker'
+categoryLayoutRenderer.propTypes = {
+  categoryContent: PropTypes.node,
+  resultsContent: PropTypes.node,
+}
```
Un component devrait commencer par une majuscule? P-e pas un standard de SUI React

alexasselin008 committed:
WIP fix unnecessary parenthesis

layershifter committed:
feat(Popup): add pinned and popperModifiers prop (#3654)

* feat(Popup): add pinned and popperModifiers prop

* add on='click'

* fix typings

* fix prop

* remove missing example, disable `keepTogether` with offset, update merge order

layershifter committed:
docs(mixed): improve perf & bundle size, fix issue with /layouts (#3652)

layershifter committed:
feat(Popup): add `popperDependencies` prop (#3657)

* feat(Popup): add `popperDependencies` prop

* Update src/modules/Popup/Popup.js

Co-Authored-By: Youngrok Kim <rok0810@gmail.com>

* add UTs, fix condition

* simplify implemention

levithomason committed:
0.87.2

levithomason committed:
docs(changelog): update changelog [ci skip]

AaronCTech committed:
docs(layouts): fix incorrectly cased tabIndex in Bootstrap layout (#3668)

The tabIndex prop was passed as tabindex (non camel-cased).

Simple and quick change :)

layershifter committed:
docs(Usage): fix typo in CRA naming

Fixes #3673.

layershifter committed:
chore(docs): make docs working in IE11 (#3678)

layershifter committed:
feat(Label): add `prompt` prop (#3680)

bubunyo committed:
feat(Form): make `error` prop shorthand in Form.Field (#3603)

* feat(Ref): add support for error messages in <Form.Field/> (Semantic-Org#3405)

* feat(Ref): add support for error messages in <Form.Field/> (#3602)

* fix(reff) update error proptype to show to to reflect in docs

* use Label component

* remove only

* update descriptions and types

* fix typings issues

ramanujanvs committed:
fix(Popup): update `hideOnScroll` description in typings (#3682)

- hideOnScroll had an incorrect description that pops up in IDEs

layershifter committed:
fix(Sidebar): fix hide animation (#3684)

* fix(Sidebar): fix hide animation

* fix props usage

* simplify code

eyas2014 committed:
fix(Dropdown): use `placeholder` as default text (#3586)

* Fix #3581

* Update Dropdown.js

* Update Dropdown-test.js

layershifter committed:
fix(Modal): support IE11, fix scrolling glitches (#3679)

* fix(Modal): support IE11, fix scrolling glitches

* fix UT

* add missing UT for StepContent

* add missing UT for NodeRegistry

* add missing UT for GridColumn

* add missing UT for PortalInner

* add missing UT for getLegacyStyles

* add missing UT for canFit

layershifter committed:
test(Checkbox): remove incorrect event combinations (#3691)

layershifter committed:
fix(Checkbox): handle click properly on label (#3692)

layershifter committed:
fix(Checkbox): handle click on root element (#3693)

levithomason committed:
0.87.3

levithomason committed:
docs(changelog): update changelog [ci skip]

layershifter committed:
docs(Composition|Shorthand): improve pages (#3707)

* docs(Composition|Shorthand): improve pages

* Update Augmentation.mdx

* Update Augmentation.mdx

alexasselin008 committed:
added proptypes to the search component

alexasselin008 committed:
forgot to put the props optional

layershifter committed:
feat(Popup): add `eventsEnabled` prop (#3715)

* feat(Popup): add `eventsEnabled` prop

* fix UT

* fix styles on Knobs

layershifter committed:
docs(examples): remove outdated Webpack examples (#3729)

layershifter committed:
style(package): format with Prettier (#3730)

layershifter committed:
chore(AutoControlledComponent): align API of trySetState() with React's (#3731)

* chore(AutoControlledComponent): align API of trySetState() with React's

* remove failing test

jongsue committed:
fix(Dropdown): retains focus after selection (#3452)

* fix(Dropdown): retains focus after selection

* remove the state argument in trySetState method to match React's API

* add assertions for 'focus' state

* update Dropdown.js

* restore style

* restore style

* use _.noop instead

layershifter committed:
fix(Checkbox): preventDefault() in mouseDown should prevent focus (#3732)

ablakey committed:
docs(Search): fix wrong description for `input` shorthand (#3727)

* Fix wrong documentation for Search `input`.

* Update Search.d.ts

ecnaidar committed:
fix(Popup): ignore clicks when `on` prop is set to `hover` only (#3667)

layershifter committed:
breaking(package): require React 16.8 (#3733)

nelsonleite committed:
feat(Dropdown): allow selecting an item when pressing the spacebar (#3702)

Due to accessibility matters (and similarity to the native <select>), this adds the possibility to select an item via Spacebar and not only with Enter

jmyrland committed:
chore(mixed): renamed unsafe lifecycle methods (#3747)

* chore(mixed): renamed unsafe lifecycle methods

Executed via `npx react-codemod rename-unsafe-lifecycles --parser babel`
Disabled lint issues for updated lines.

Closes #3741
Related to #2732

* bump react to ensure that CI will pass

* disable broken UTs

ablakey-locusrobotics committed:
docs(Icon): fix a typo (#3749)

layershifter committed:
chore(lib): remove `customPropTypes.as` (#3751)

layershifter committed:
docs(Usage|Theming): improve pages, introduce CLI util (#3752)

b-smets committed:
fix(typings): add missing `StrictMenuProps` type export (#3755)

SandipNirmal committed:
docs(Popup): add example for `delay` prop in Popup (#3754)

* Docs(Popup): Added example to use delay with Popup.

Added new example to use show/hide delay with the Popup component.

* Update PopupExampleDelay.js

layershifter committed:
docs(CodeSandbox): fix paths for images in exported examples (#3758)

levithomason committed:
0.88.0

levithomason committed:
docs(changelog): update changelog [ci skip]

layershifter committed:
docs(Embed): fix example with custom URL

layershifter committed:
docs(Theming): use @semantic-ui-react/craco-less plugin (#3767)

fxOne committed:
feat(Popup): add `positionFixed` to support fixed mode in `Popper.JS` (#3760)

* feat(Popup): add positionFixed to support popper fixed mode (#3718)

Add support for popper.js to position the popup correctly if used with a mountNode inside a `position: fixed` container

* feat(Popup): add `positionFixed` to support fixed mode in `Popper.JS`

Simplify code

* Update Popup-test.js

* Update Popup.d.ts

* Update Popup.js

honzajerabek committed:
fix(Dropdown): closing the search menu on spacebar press (#3766)

* Fix closing the search menu on spacebar press

* Update Dropdown.js

layershifter committed:
chore(Ref): use component from Stardust UI (#3774)

* chore(Ref): use component from Stardust

* chore(Ref): use component from Stardust

* fix UT

layershifter committed:
fix(Sidebar): handle clicks properly (#3772)

* fix(Sidebar): handle clicks properly

* fix lock

levithomason committed:
0.88.1

levithomason committed:
docs(changelog): update changelog [ci skip]

layershifter committed:
chore(lib): add ModernAutoControlledComponent (#3776)

aoelen committed:
feat(FormField): make form field error accessible (#3822)

* feat(FormField): make form field error accessible

* Change type of id on FormField

* Add alert role on error labels and include error example

unbug committed:
fix(Search): use result.id for SearchResult key (#3848)

* Fixed key value for SearchResult Component

* Remove result.key

chrisbull committed:
feat(TextArea): export StrictTextAreaProps typing (#3846)

yuuyu00 committed:
docs(misc): fix typos (#3837)

* docs(CONTRIBUTING): Fix typo

* docs(ComponentExample): Fix typo

* docs(Types/index.js): Fix typo

anupamasok committed:
docs: fix minor grammar and punctuation errors (#3818)

ridvankaradag committed:
fix(FeedSummary): add missing space around content (#3836)

* missing space

* wrap feed summary content shorthand in whitespace

* test feed summary content whitespace

spencerbyw committed:
docs: update grammar/wording for tab pane shorthand (#3801)

levithomason committed:
chore: remove stale bot (#3858)

Suzi004 committed:
docs(MenuExampleHeaderVertical): fix onClick handler (#3799)

handleItemClick = (e, { name }) => this.setState({ activeItem: name });

levithomason committed:
chore: update husky hooks (#3859)

chrisbrainerd committed:
docs: update warning about using yarn (#3816)

levithomason committed:
remove v2 in example title

levithomason committed:
Merge branch 'master' into feat/3630_Custom_SearchCategory_Renderer
